src/app/SharpRaven/Data/HttpRequester.cs

```diff
@@ -139,21 +120,26 @@ public RequestData Data
         }
 
         /// <summary>
-        /// Gets the <see cref="JsonPacket"/> being sent to Sentry.
+        /// Gets the <see cref="HttpWebRequest"/> instance being used to perform the HTTP request to Sentry.
         /// </summary>
-        public JsonPacket Packet
+        public HttpWebRequest WebRequest
         {
-            get { return this.packet; }
+            get { return this.webRequest; }
         }
 
         /// <summary>
-        /// Gets the <see cref="HttpWebRequest"/> instance being used to perform the HTTP request to Sentry.
+        /// Uses the event identifier on the JsonPacket if there is one.
         /// </summary>
-        public HttpWebRequest WebRequest
+        /// <param name="eventId">The event identifier.</param>
+        public void UseEventId(string eventId)
```

Comment:
Is the primary purpose of this method for testing? Is it worth making this functionality work for the queue implementation also?

Reply:
The latter. The idea was that the QueuedRequester could generate the Id and set the payload for that Id and return it.